<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.40">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Benchmark – Damage Benchmark</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-549806ee2085284f45b00abea8c6df48.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-530bf8e59df25186f8bffdc63bad3503.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Damage Benchmark</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./benchmark.html" aria-current="page"> 
<span class="menu-text">Benchmark</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./data.html"> 
<span class="menu-text">Datasets</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./results.html"> 
<span class="menu-text">Results</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./contribute.html"> 
<span class="menu-text">Contribute</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
    <div class="dropdown">
      <a href="" title="" id="quarto-navigation-tool-dropdown-0" class="quarto-navigation-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false" role="link" aria-label=""><i class="bi bi-github"></i></a>
      <ul class="dropdown-menu" aria-labelledby="quarto-navigation-tool-dropdown-0">
          <li>
            <a class="dropdown-item quarto-navbar-tools-item" href="https://github.com/AlicenJoyHenning/Damage-Benchmark">
            Source Code
            </a>
          </li>
          <li>
            <a class="dropdown-item quarto-navbar-tools-item" href="https://github.com/AlicenJoyHenning/Damage-Benchmark/issues">
            Report a Bug
            </a>
          </li>
      </ul>
    </div>
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#strategies" id="toc-strategies" class="nav-link active" data-scroll-target="#strategies">Strategies</a>
  <ul>
  <li><a href="#tool-based-strategies" id="toc-tool-based-strategies" class="nav-link" data-scroll-target="#tool-based-strategies">Tool-based strategies</a>
  <ul>
  <li><a href="#ddqc" id="toc-ddqc" class="nav-link" data-scroll-target="#ddqc">1. ddqc</a></li>
  <li><a href="#dropletqc" id="toc-dropletqc" class="nav-link" data-scroll-target="#dropletqc">2. DropletQC</a></li>
  <li><a href="#ensemblekqc" id="toc-ensemblekqc" class="nav-link" data-scroll-target="#ensemblekqc">3. EnsembleKQC</a></li>
  <li><a href="#miqc" id="toc-miqc" class="nav-link" data-scroll-target="#miqc">4. miQC</a></li>
  <li><a href="#sampleqc" id="toc-sampleqc" class="nav-link" data-scroll-target="#sampleqc">5. SampleQC</a></li>
  <li><a href="#scater" id="toc-scater" class="nav-link" data-scroll-target="#scater">6. Scater</a></li>
  <li><a href="#validrops" id="toc-validrops" class="nav-link" data-scroll-target="#validrops">7. valiDrops</a></li>
  </ul></li>
  <li><a href="#manual-strategies" id="toc-manual-strategies" class="nav-link" data-scroll-target="#manual-strategies">Manual strategies</a>
  <ul>
  <li><a href="#fixed-thresholding" id="toc-fixed-thresholding" class="nav-link" data-scroll-target="#fixed-thresholding">Fixed thresholding</a>
  <ul>
  <li><a href="#mitochondrial-percent-expression" id="toc-mitochondrial-percent-expression" class="nav-link" data-scroll-target="#mitochondrial-percent-expression">8. Mitochondrial percent expression</a></li>
  </ul></li>
  <li><a href="#adaptive-thresholding" id="toc-adaptive-thresholding" class="nav-link" data-scroll-target="#adaptive-thresholding">Adaptive thresholding</a>
  <ul>
  <li><a href="#mitochondrial-percent-expression-1" id="toc-mitochondrial-percent-expression-1" class="nav-link" data-scroll-target="#mitochondrial-percent-expression-1">9. Mitochondrial percent expression</a></li>
  <li><a href="#mitochondrial-ribosomal-percent-expression" id="toc-mitochondrial-ribosomal-percent-expression" class="nav-link" data-scroll-target="#mitochondrial-ribosomal-percent-expression">10. Mitochondrial &amp; ribosomal percent expression</a></li>
  <li><a href="#mitochondrial-ribosomal-percent-expression-library-size" id="toc-mitochondrial-ribosomal-percent-expression-library-size" class="nav-link" data-scroll-target="#mitochondrial-ribosomal-percent-expression-library-size">11. Mitochondrial, ribosomal percent expression &amp; library size</a></li>
  <li><a href="#library-size" id="toc-library-size" class="nav-link" data-scroll-target="#library-size">12. Library size</a></li>
  <li><a href="#malat1-percent-expression" id="toc-malat1-percent-expression" class="nav-link" data-scroll-target="#malat1-percent-expression">13. MALAT1 percent expression</a></li>
  <li><a href="#malat1-mitochondrial-percent-expression" id="toc-malat1-mitochondrial-percent-expression" class="nav-link" data-scroll-target="#malat1-mitochondrial-percent-expression">14. MALAT1 &amp; mitochondrial percent expression</a></li>
  </ul></li>
  </ul></li>
  </ul></li>
  <li><a href="#metrics" id="toc-metrics" class="nav-link" data-scroll-target="#metrics">Metrics</a>
  <ul>
  <li><a href="#performance-metrics" id="toc-performance-metrics" class="nav-link" data-scroll-target="#performance-metrics">Performance metrics</a></li>
  <li><a href="#performance-agnostic-metrics" id="toc-performance-agnostic-metrics" class="nav-link" data-scroll-target="#performance-agnostic-metrics">Performance agnostic metrics</a>
  <ul>
  <li><a href="#damage-focus" id="toc-damage-focus" class="nav-link" data-scroll-target="#damage-focus">Damage focus</a></li>
  <li><a href="#stringency" id="toc-stringency" class="nav-link" data-scroll-target="#stringency">Stringency</a></li>
  <li><a href="#consistency" id="toc-consistency" class="nav-link" data-scroll-target="#consistency">Consistency</a></li>
  <li><a href="#usability" id="toc-usability" class="nav-link" data-scroll-target="#usability">Usability</a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Benchmark</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="strategies" class="level2">
<h2 class="anchored" data-anchor-id="strategies">Strategies</h2>
<section id="tool-based-strategies" class="level3">
<h3 class="anchored" data-anchor-id="tool-based-strategies">Tool-based strategies</h3>
<section id="ddqc" class="level4">
<h4 class="anchored" data-anchor-id="ddqc">1. ddqc</h4>
<p><strong>Article</strong>: <a href="https://doi.org/10.1186/s13059-022-02820-w">https://doi.org/10.1186/s13059-022-02820-w</a></p>
<p><a href="https://github.com/ayshwaryas/ddqc"><img src="https://img.shields.io/badge/github-%23121011.svg?style=for-the-badge&amp;logo=github&amp;logoColor=white" class="img-fluid" alt="GitHub"></a> <a href="https://github.com/ayshwaryas/ddqc"><img src="https://img.shields.io/github/stars/ayshwaryas/ddqc?style=social.png" class="img-fluid" alt="GitHub stars"></a> <img src="https://img.shields.io/github/last-commit/ayshwaryas/ddqc.png" class="img-fluid" alt="Last Commit"></p>
<p>Identifies damaged cells within sample-specific clusters using the Louvain algorithm for clustering and MAD-based univariate outlier detection of UMI, feature counts, and mitochondrial or ribosomal percentages.</p>
<p><a href="https://github.com/ayshwaryas/ddqc/blob/master/tutorials/ddqc_tutorial.ipynb">Tutorial</a></p>
</section>
<section id="dropletqc" class="level4">
<h4 class="anchored" data-anchor-id="dropletqc">2. DropletQC</h4>
<p><strong>Article</strong>: <a href="https://doi.org/10.1186/s13059-021-02547-0">https://doi.org/10.1186/s13059-021-02547-0</a></p>
<p><a href="https://github.com/powellgenomicslab/DropletQC"><img src="https://img.shields.io/badge/github-%23121011.svg?style=for-the-badge&amp;logo=github&amp;logoColor=white" class="img-fluid" alt="GitHub"></a> <a href="https://github.com/powellgenomicslab/DropletQC"><img src="https://img.shields.io/github/stars/powellgenomicslab/DropletQC?style=social.png" class="img-fluid" alt="GitHub stars"></a> <img src="https://img.shields.io/github/last-commit/powellgenomicslab/DropletQC.png" class="img-fluid" alt="Last Commit"></p>
<p>Uses nuclear fraction (ratio of unspliced to spliced transcripts) and UMI counts with an Expectation-Maximization model to classify damaged cells.</p>
<p><a href="https://powellgenomicslab.github.io/DropletQC/articles/DropletQC.html#identifying-damaged-cells-1">Tutorial</a></p>
</section>
<section id="ensemblekqc" class="level4">
<h4 class="anchored" data-anchor-id="ensemblekqc">3. EnsembleKQC</h4>
<p><strong>Article</strong>: <a href="https://doi.org/10.1007/978-3-030-26969-2_47">https://doi.org/10.1007/978-3-030-26969-2_47</a></p>
<p><a href="https://github.com/mzhq/EnsembleKQC"><img src="https://img.shields.io/badge/github-%23121011.svg?style=for-the-badge&amp;logo=github&amp;logoColor=white" class="img-fluid" alt="GitHub"></a> <a href="https://github.com/mzhq/EnsembleKQC"><img src="https://img.shields.io/github/stars/mzhq/EnsembleKQC?style=social.png" class="img-fluid" alt="GitHub stars"></a> <img src="https://img.shields.io/github/last-commit/mzhq/EnsembleKQC.png" class="img-fluid" alt="Last Commit"></p>
<p>Combines multiple quality control measures to detect damaged cells, focusing on ensemble-based metrics</p>
</section>
<section id="miqc" class="level4">
<h4 class="anchored" data-anchor-id="miqc">4. miQC</h4>
<p><strong>Article</strong>: <a href="https://doi.org/10.1371/journal.pcbi.1009290">https://doi.org/10.1371/journal.pcbi.1009290</a></p>
<p><a href="https://github.com/greenelab/miQC"><img src="https://img.shields.io/badge/github-%23121011.svg?style=for-the-badge&amp;logo=github&amp;logoColor=white" class="img-fluid" alt="GitHub"></a> <a href="https://github.com/greenelab/miQC"><img src="https://img.shields.io/github/stars/greenelab/miQC?style=social.png" class="img-fluid" alt="GitHub stars"></a> <img src="https://img.shields.io/github/last-commit/greenelab/miQC.png" class="img-fluid" alt="Last Commit"></p>
<p>Applies a mixture model to mitochondrial percentage and library complexity to assign posterior probabilities for identifying damaged cells.</p>
<p><a href="https://www.bioconductor.org/packages/release/bioc/vignettes/miQC/inst/doc/miQC.html">Tutorial</a></p>
</section>
<section id="sampleqc" class="level4">
<h4 class="anchored" data-anchor-id="sampleqc">5. SampleQC</h4>
<p><strong>Article</strong>: <a href="https://doi.org/10.1186/s13059-023-02859-3">https://doi.org/10.1186/s13059-023-02859-3</a></p>
<p><a href="https://github.com/wmacnair/SampleQC"><img src="https://img.shields.io/badge/github-%23121011.svg?style=for-the-badge&amp;logo=github&amp;logoColor=white" class="img-fluid" alt="GitHub"></a> <a href="https://github.com/wmacnair/SampleQC"><img src="https://img.shields.io/github/stars/wmacnair/SampleQC?style=social.png" class="img-fluid" alt="GitHub stars"></a> <img src="https://img.shields.io/github/last-commit/wmacnair/SampleQC.png" class="img-fluid" alt="Last Commit"></p>
<p>Automates sample-wide multivariate outlier detection using robust Mahalanobis distances for mitochondrial, ribosomal, and UMI counts.</p>
<p><a href="https://github.com/wmacnair/SampleQC/blob/main/README.md">Tutorial</a></p>
</section>
<section id="scater" class="level4">
<h4 class="anchored" data-anchor-id="scater">6. Scater</h4>
<p><strong>Article</strong>: <a href="https://doi.org/10.1093/bioinformatics/btw777">https://doi.org/10.1093/bioinformatics/btw777</a></p>
<p><a href="https://github.com/alanocallaghan/scater"><img src="https://img.shields.io/badge/github-%23121011.svg?style=for-the-badge&amp;logo=github&amp;logoColor=white" class="img-fluid" alt="GitHub"></a> <a href="https://github.com/alanocallaghan/scater"><img src="https://img.shields.io/github/stars/alanocallaghan/scater?style=social.png" class="img-fluid" alt="GitHub stars"></a> <img src="https://img.shields.io/github/last-commit/alanocallaghan/scater.png" class="img-fluid" alt="Last Commit"></p>
<p>Provides diagnostic visualizations and filtering functions to assess and remove damaged cells using user-defined thresholds</p>
<p><a href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-scater-qc/tutorial.html#automatic-pca-filtering">Tutorial</a></p>
</section>
<section id="validrops" class="level4">
<h4 class="anchored" data-anchor-id="validrops">7. valiDrops</h4>
<p><strong>Article</strong>: <a href="https://doi.org/10.1101/2023.02.07.526574">https://doi.org/10.1101/2023.02.07.526574</a></p>
<p><a href="https://github.com/madsen-lab/valiDrops"><img src="https://img.shields.io/badge/github-%23121011.svg?style=for-the-badge&amp;logo=github&amp;logoColor=white" class="img-fluid" alt="GitHub"></a><br>
<a href="https://github.com/madsen-lab/valiDrops"><img src="https://img.shields.io/github/stars/madsen-lab/valiDrops?style=social.png" class="img-fluid" alt="GitHub stars"></a> <img src="https://img.shields.io/github/last-commit/madsen-lab/valiDrops.png" class="img-fluid" alt="Last Commit"></p>
<p>Detects apoptotic cells using scaled and transformed QC features with logistic regression and cross-validation for soft labeling.</p>
<hr>
</section>
</section>
<section id="manual-strategies" class="level3">
<h3 class="anchored" data-anchor-id="manual-strategies">Manual strategies</h3>
<section id="fixed-thresholding" class="level4">
<h4 class="anchored" data-anchor-id="fixed-thresholding">Fixed thresholding</h4>
<section id="mitochondrial-percent-expression" class="level5">
<h5 class="anchored" data-anchor-id="mitochondrial-percent-expression">8. Mitochondrial percent expression</h5>
<p>Involves setting a fixed value for mitochondrial percent expression above which cells are classified as damaged. Implementation of this method varies based on preferred platform, for example using default functions in the <a href="https://satijalab.org/seurat/"><code>Seurat</code></a> R package,</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>data[[<span class="st">"percent.mt"</span>]] <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(data, <span class="at">pattern =</span> <span class="st">"^MT-"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">subset</span>(data, <span class="at">subset =</span> percent.mt <span class="sc">&lt;</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>And the <a href="https://scanpy.readthedocs.io/en/stable/"><code>Scanpy</code></a> python package,</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>data.var[<span class="st">'mt'</span>] <span class="op">=</span> data.var_names.<span class="bu">str</span>.startswith(<span class="st">'MT-'</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>sc.pp.calculate_qc_metrics(data, qc_vars<span class="op">=</span>[<span class="st">'mt'</span>], percent_top<span class="op">=</span><span class="va">None</span>, log1p<span class="op">=</span><span class="va">False</span>, inplace<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>data <span class="op">=</span> data[data.obs.pct_counts_mt <span class="op">&lt;</span> <span class="dv">10</span>, :]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br></p>
</section>
</section>
<section id="adaptive-thresholding" class="level4">
<h4 class="anchored" data-anchor-id="adaptive-thresholding">Adaptive thresholding</h4>
<p>Thresholds set based on the dataset’s distribution of a certain metric using a simple percentile-based method. Upper thresholds (e.g., for mitochondrial and MALAT1 percent expression) were set at the 90th percentile, while lower thresholds (e.g., for ribosomal percent expression and library size) were set at the 10th percentile. Note: library size refers to an equally weighted consideration of the log10 transformed UMI and feature counts.</p>
<section id="mitochondrial-percent-expression-1" class="level5">
<h5 class="anchored" data-anchor-id="mitochondrial-percent-expression-1">9. Mitochondrial percent expression</h5>
<p>Filtering cells with</p>
<ul>
<li>mitochondrial percent expression above the 90th percentile.</li>
</ul>
</section>
<section id="mitochondrial-ribosomal-percent-expression" class="level5">
<h5 class="anchored" data-anchor-id="mitochondrial-ribosomal-percent-expression">10. Mitochondrial &amp; ribosomal percent expression</h5>
<p>Filtering cells with</p>
<ul>
<li>mitochondrial percent expression above the 90th percentile AND</li>
<li>with ribosomal percent expression below the 10th percentile.</li>
</ul>
</section>
<section id="mitochondrial-ribosomal-percent-expression-library-size" class="level5">
<h5 class="anchored" data-anchor-id="mitochondrial-ribosomal-percent-expression-library-size">11. Mitochondrial, ribosomal percent expression &amp; library size</h5>
<p>Filtering cells with</p>
<ul>
<li>mitochondrial percent expression above the 90th percentile,</li>
<li>ribosomal percent expression below the 10th percentile AND</li>
<li>library size below the 10th percentile.</li>
</ul>
</section>
<section id="library-size" class="level5">
<h5 class="anchored" data-anchor-id="library-size">12. Library size</h5>
<p>Filtering cells with</p>
<ul>
<li>library sizes below the 10th percentile.</li>
</ul>
</section>
<section id="malat1-percent-expression" class="level5">
<h5 class="anchored" data-anchor-id="malat1-percent-expression">13. MALAT1 percent expression</h5>
<p>Filtering cells with</p>
<ul>
<li>MALAT1 percent expression above the 90th percentile.</li>
</ul>
</section>
<section id="malat1-mitochondrial-percent-expression" class="level5">
<h5 class="anchored" data-anchor-id="malat1-mitochondrial-percent-expression">14. MALAT1 &amp; mitochondrial percent expression</h5>
<p>Filtering cells with</p>
<ul>
<li>MALAT1 percent expression above the 90th percentile AND</li>
<li>mitochondrial percent expression above the 90th percentile.</li>
</ul>
<p><br> <br></p>
<hr>
</section>
</section>
</section>
</section>
<section id="metrics" class="level2">
<h2 class="anchored" data-anchor-id="metrics">Metrics</h2>
<section id="performance-metrics" class="level3">
<h3 class="anchored" data-anchor-id="performance-metrics">Performance metrics</h3>
<p><span class="math display">\[
\text{Performance} = \text{Precision} + \text{PR-AUC} + \text{Jaccard Index} + \text{F1 Score}
\]</span></p>
<p>where,</p>
<ul>
<li><p><strong>Precision</strong> <span class="math inline">\(\text{(0, 1)}\)</span> Of all damaged predictions made, what proportion were true damaged cells?<br>
<em>Median score taken from the 5 experimentally verified ground truth datasets.</em></p></li>
<li><p><strong>PR-AUC</strong> <span class="math inline">\(\text{(0, 1)}\)</span> Of all predictions, were damaged cells correctly and comprehensively identified (i.e., were true damaged cells missed)?<br>
<em>Median score taken from the 5 experimentally verified ground truth datasets.</em></p></li>
<li><p><strong>Jaccard Index</strong> <span class="math inline">\(\text{(0, 1)}\)</span> How similar were the filtered datasets to the positive, damage-free control (closer to the control indicates better correction for damage)?<br>
<em>Median score taken from the 30 simulated ground truth datasets.</em></p></li>
<li><p><strong>F1 Score</strong> <span class="math inline">\(\text{(0, 1)}\)</span> Were the differentially expressed genes of the filtered datasets correct compared to the positive, damage-free control?<br>
<em>Median score taken from the 30 simulated ground truth datasets.</em></p></li>
</ul>
<p>Thus, <strong>Performance</strong> lies in the range <span class="math inline">\(\text{(0, 4)}\)</span>, with 4 being the highest performance possible.</p>
<p>You can view the individual performance scores for the strategies, <a href="">here</a>.</p>
<hr>
</section>
<section id="performance-agnostic-metrics" class="level3">
<h3 class="anchored" data-anchor-id="performance-agnostic-metrics">Performance agnostic metrics</h3>
<section id="damage-focus" class="level4">
<h4 class="anchored" data-anchor-id="damage-focus">Damage focus</h4>
<p>This score measures how many of the damaged predictions originate from the population of damaged cells that have a heavily altered transcriptomic profile compared to control cells. Here, a strategy that is highly specific to this population, rather than to the population of damaged cells that appear identical to control cells and pose no contamination risk, is ideal. This value, derived from the median across the 5 experimentally verified ground truth datasets, is a proportion and lies in the range <span class="math inline">\(\text{(0, 1)}\)</span> where we suggest that closer to 1 is ideal.</p>
<p>You can view the isolated damage focus scores, <a href="damage-focus.html">here</a>.</p>
<p><br></p>
</section>
<section id="stringency" class="level4">
<h4 class="anchored" data-anchor-id="stringency">Stringency</h4>
<p>This score measures how many of the cells present were labelled as damaged by the strategy. While this percentage is not especially valuable when viewed in an isolated dataset, collecting it across many datasets, here 51, gives a good idea of how stringent the strategy is in general and how a user can expect it to be when applied to new data. This is offered in response to one of the most prominent concerns for damaged cell filtering strategies: that they will remove too many cells, i.e.&nbsp;be too stringent, or won’t remove enough, i.e.&nbsp;be too lenient. Being a percentage, this score ranges from <span class="math inline">\(\text{(0, 100)}\)</span> where closer to 100 represents more stringent strategies.</p>
<p>You can view the isolated stringency scores, <a href="stringency.html">here</a>.</p>
<p><br></p>
</section>
<section id="consistency" class="level4">
<h4 class="anchored" data-anchor-id="consistency">Consistency</h4>
<p>This score measures the deviation in the output of a tool across datasets,</p>
<p><span class="math display">\[
\text{Consistency} = 1 - \left( \text{c}_1 \cdot \text{similarity deviation} + \text{c}_2 \cdot \text{stringency deviation} \right)
\]</span></p>
<p>made up of two components, <span class="math inline">\(\text{similarity deviation}\)</span> and <span class="math inline">\(\text{stringency deviation}\)</span>, combined by constants <span class="math inline">\(c_{1}\)</span> and <span class="math inline">\(c_{2}\)</span> which are both <span class="math inline">\(0.5\)</span> but can be altered to emphasize’s ones contribution over the other.</p>
<p>The <span class="math inline">\(\text{similarity deviation}\)</span> for a given strategy <span class="math inline">\(i\)</span> is computed as the mean of the standard deviations of Cohen’s kappa scores <span class="math inline">\(\kappa_{ij}\)</span> for strategy <span class="math inline">\(i\)</span> compared to another strategy <span class="math inline">\(j\)</span> <span class="math inline">\((j \neq i)\)</span> across datasets. Here, <span class="math inline">\(\kappa_{ij}\)</span> is calculated by comparing the set of damaged cell labels given by strategy <span class="math inline">\(i\)</span> to those given by strategy <span class="math inline">\(j\)</span> for the same dataset. The standard deviation <span class="math inline">\(\sigma(\kappa_{ij})\)</span> is taken across all datasets, and the mean is calculated over the all deviation scores for tool <span class="math inline">\(i\)</span>, amounting to <span class="math inline">\({(J - 1)}\)</span> where <span class="math inline">\(J\)</span> is the total number of strategies. Since <span class="math inline">\(\kappa\)</span> ranges from <span class="math inline">\(-1\)</span> to <span class="math inline">\(1\)</span>, where <span class="math inline">\(1\)</span> indicates complete agreement and <span class="math inline">\(-1\)</span> indicates complete disagreement, deviation scores typically range on the lower end of <span class="math inline">\((0, 1)\)</span>. To standardize the similarity deviation score for the consistency equation, min-max scaling is applied across strategies, setting the highest mean deviation to <span class="math inline">\(1\)</span> and the lowest to <span class="math inline">\(0\)</span>.</p>
<p><span class="math display">\[
\text{similarity deviation} = \frac{1}{J - 1} \sum_{j \neq i} \sigma(\kappa_{ij})
\]</span></p>
<p>More simply, <span class="math inline">\(\text{stringency deviation}\)</span> is taken as the standard deviation in the stringency score for a particular strategy, <span class="math inline">\(i\)</span>, across datasets, where stringency is the percentage of damaged cells detected by the strategy. And as with <span class="math inline">\(\text{similarity deviation}\)</span>, the <span class="math inline">\(\text{stringency deviation}\)</span> score is standardised by min-max scaling for the consistency equation, setting the highest deviation to <span class="math inline">\(1\)</span> and the lowest to <span class="math inline">\(0\)</span>.</p>
<p><span class="math display">\[
\text{stringency deviation} =  \sigma(\text{damaged cells / total cells})
\]</span></p>
<p><br></p>
<p>Note, each component is measured across the dataset types in isolation before being combined in the form of a median to generate the final deviation score. i.e, <span class="math inline">\(\text{stringency deviation}\)</span> is the median of the score taken across the groundtruth, non-groundtruth, and simulated datasets. This is done to avoid imbalancing the consistency score with any one dataset type.</p>
<p><br></p>
<p>You can view the isolated consistency scores, <a href="consistency.html">here</a>.</p>
<p><br> <br></p>
</section>
<section id="usability" class="level4">
<h4 class="anchored" data-anchor-id="usability">Usability</h4>
<p>This is a manual and automated scoring of tool-based strategies according to 6 criteria,</p>
<table class="caption-top table">
<colgroup>
<col style="width: 50%">
<col style="width: 19%">
<col style="width: 7%">
<col style="width: 8%">
<col style="width: 13%">
</colgroup>
<thead>
<tr class="header">
<th>Criterion</th>
<th>Automated</th>
<th></th>
<th>No</th>
<th>Yes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Presence of documentation</td>
<td></td>
<td>0</td>
<td>1</td>
<td>1</td>
</tr>
<tr class="even">
<td>More than one code deployment</td>
<td>✓</td>
<td>0</td>
<td>1</td>
<td>1</td>
</tr>
<tr class="odd">
<td>Last adjustment to tool less than two years ago</td>
<td>✓</td>
<td>0</td>
<td>1</td>
<td>1</td>
</tr>
<tr class="even">
<td>Alterations to core code required</td>
<td></td>
<td>1</td>
<td>0</td>
<td>1</td>
</tr>
<tr class="odd">
<td>GitHub stars<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a></td>
<td>✓</td>
<td>0</td>
<td>…</td>
<td>5</td>
</tr>
<tr class="even">
<td>Ease of use<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a></td>
<td></td>
<td>…</td>
<td>…</td>
<td>5</td>
</tr>
</tbody>
</table>
<p>You can view the isolated usability scores, <a href="usability.html">here</a>.</p>
<p><br></p>
<hr>
<p>To add a new metric or suggest an adjustment to the above metrics, please visit the <a href="contribute.html">Contribute</a> page.</p>
<hr>


</section>
</section>
</section>


<div id="quarto-appendix" class="default"><section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes"><h2 class="anchored quarto-appendix-heading">Footnotes</h2>

<ol>
<li id="fn1"><p>GitHub stars underwent min-max scaling.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn2"><p>Subjective score based on installation, dependencies, number of steps required to obtain output classification, accessibility of input data information, etc.<a href="#fnref2" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">
<p>Built with <a href="https://quarto.org/">Quarto</a></p>
</div>
  </div>
</footer>




</body></html>